### PR TITLE
Refactored item hashtable code

### PR DESCRIPTION
**Pull request by ericyc16**
Addresses issue #703 

This pull request refactors code involving item hashtables to reduce duplicate code. This was done in preparation of implementing item stacking, as described in the issue above.

The following functions were added to the item module:

- `int add_item_to_hash(item_hash_t **ht, item_t *new_item);`
- `item_list_t *get_all_items_in_hash(item_hash_t **ht);`
- `int remove_item_from_hash(item_hash_t **ht, item_t *old_item);`

Functions in the game, room, player, and npc module were changed to utilize these functions. Additionally, `remove_item_from_room/player/npc` functions were added to the appropriate modules. 

Finally, tests were written to ensure that refactoring this code didn't break anything.

One final note: my code editor automatically converts hard tabs to 4 spaces. Apparently, there were a bunch of hard tabs in the `test_items.c` and `test_rooms.c` files. So there aren't really a whopping 556 additions and 360 deletions in this PR; a decent amount of that is just converting tabs to spaces.

### CODE REVIEW

File: tests/npc/test_npc.c [hunk 1]
```diff
@@ -149,5 +149,22 @@ Test(npc, add_item_to_npc)
...
+    npc_t *npc = npc_new("npc", 100, NULL);
+    item_t *test_item = item_new("item", "short", "long");
+    int rc;
+    
+    rc = add_item_to_npc(npc, test_item);
+    cr_assert_eq(rc, SUCCESS, "add_item_to_npc failed to "
+                 "add an item to npc");
+    
+    rc = remove_item_from_npc(npc, test_item);
+    printf("%d\n", rc);
```
Comment (lilyehsani) [inline tests/npc/test_npc.c:167]: I don't think this line is necessary

Comment (ericyc16) [reply tests/npc/test_npc.c]: Should be gone now!

File: include/game-state/item.h [hunk 2]
```diff
@@ -91,6 +91,40 @@ char *get_sdesc_item(item_t *item);
  */
 char *get_ldesc_item(item_t *item);
 
+/* Adds an item to a hashtable of items
+ * 
+ * Parameters:
+ *  pointer to hashtable of items (pointer necessary for uthash to work)
+ *  item to add to hashtable
+ *
```
Comment (MaxineK36) [inline include/game-state/item.h:99]: What's the behavior if the item is already in the hashtable?

File: include/game-state/item.h [hunk 3]
```diff
@@ -91,6 +91,40 @@ char *get_sdesc_item(item_t *item);
...
+
+/* Removes an item from a hashtable of items
+ * Note that the memory associated with this item is not freed
+ * 
+ * Parameters:
+ *  pointer to hashtable of items (pointer necessary for uthash to work)
+ *  item to add to hashtable
+ * 
+ * Return:
+ *  SUCCESS if successful, FAILURES if failed
```
Comment (MaxineK36) [inline include/game-state/item.h:124]: *FAILURE if failed

File: src/game-state/src/item.c [hunk 4]
```diff
@@ -67,6 +67,45 @@ char *get_ldesc_item(item_t *item)
...
+        tmp = malloc(sizeof(item_list_t));
+        tmp->item = curr_item;
+        LL_APPEND(head, tmp);
+    }
+    return head;
+}
+
+/* See item.h */
+int remove_item_from_hash(item_hash_t **ht, item_t *old_item)
+{
```
Comment (MaxineK36) [inline src/game-state/src/item.c:104]: Do you know what the behavior is here if the item isn't in the hashtable?

File: tests/game-state/test_items.c [hunk 5]
```diff
@@ -77,6 +77,58 @@ return long description");
  for NULL item");
 }
 
+/* Checks that add_item_to_hash adds item to an item hashtable as expected */
```
Comment (MaxineK36) [inline tests/game-state/test_items.c:80]: Can you add a note that this also checks to see if adding twice returns FAILURE?

File: tests/game-state/test_items.c [hunk 6]
```diff
@@ -77,6 +77,58 @@ return long description");
...
+    item_t *test_item2 = item_new("item2", "short", "long");
+    
+    item_list_t *list = get_all_items_in_hash(&ht);
+    cr_assert_eq(list, NULL, "get_all_items_in_hash did not return NULL for "
+                 "empty hashtable");
+    
+    add_item_to_hash(&ht, test_item1);
+    add_item_to_hash(&ht, test_item2);
+    list = get_all_items_in_hash(&ht);
+    cr_assert_not_null(list, "get_all_items_in_hash returned NULL for "
```
Comment (MaxineK36) [inline tests/game-state/test_items.c:111]: can you loop through the LL here and make sure it has both items?

**Comment by ericyc16** [comment]
After fixing a few style issues and adding a test to ensure all items in a hashtable make it into a linked list, the code refactoring chiventure to reduce duplicate code regarding item hashtables has been merged with the dev branch. 

The issue related to this pull is still open because this PR was step 1 of the process of implementing item stacking. Another PR will be made once item stacking is actually implemented, and then the issue will be closed.

**Comment by MaxineK36** [comment]
Issue Score: ✔️++

Comments:
Great job! This PR was very cleanly written (and very helpful to chiventure).

**Pull request merged**